resources/install.rb

```diff
@@ -0,0 +1,167 @@
+resource_name :osl_php_install
+provides :osl_php_install
+unified_mode true
+
+property :packages, Array, default: []
+property :php_packages, Array, default: []
+property :use_ius, [true, false], default: false
+property :version, String
+property :use_composer, [true, false], default: false
+property :composer_version, String, default: '2.2.18'
+property :use_opcache, [true, false], default: false
+property :opcache_conf, Hash, default: {}
+
+action :install do
+  system_php = new_resource.version.nil?
+  version = new_resource.version || php_version
+  shortver = version.delete('.') # version X.X -> XX
+  all_packages = new_resource.packages.to_a
+  all_php_packages = new_resource.php_packages.to_a
+
+  # To avoid warnings about including recipes in a resource, do the same things these recipes do ---
+  # include_recipe 'osl-selinux'
+  selinux_install 'osl-selinux'
+  selinux_state 'osl-selinux' do
+    action :enforcing
+  end
+
+  # include_recipe 'osl-repos::epel'
+  osl_repos_epel 'default' do
+    exclude %w(php73* php74*) if version == '7.2'
```

Comment:
I remember there being an issue with conflicting versions if this wasn't excluded; it would try to install the latest version (7.4) instead of the specified one.

Reply:
Huh you might be right; I just tried running `packages-php72-centos-7` with the recipe and no excludes and it converged fine... I'll try with a few more suites and see if there are any other issues.

Reply:
Yep seems like this is working! I've switched this back to using the recipe.